### ISSUE TITLE
Rule idea: `no-useless-constructor`

### ISSUE THREAD

**Issue by silverwind** [ota-meshi/eslint-plugin-regexp#783]
**Motivation**
Saw this abuse in the wild.

**Description**
Avoid useless `RegExp` constructor around literals.

**Examples**

```js
/* ✗ BAD */
var foo = new RegExp(/regex/)

/* ✓ GOOD */
var foo = /regex/
```

**Comment by ota-meshi** [comment]
Thanks for the rule suggestion.
However, is it different from the `prefer-regex-literals` rule in ESLint core?

https://eslint.org/docs/latest/rules/prefer-regex-literals#disallowredundantwrapping

**Comment by silverwind** [comment]
It is the same as `disallowRedundantWrapping`, I wasn't aware of that option. Added it to my config. Thanks for pointing it out.
